projects/plugins/boost/app/assets/src/js/features/image-size-analysis/recommendations-meta/recommendations-meta.tsx

```diff
@@ -1,7 +1,238 @@
-import { Notice } from '@automattic/jetpack-components';
+import { useState, useEffect } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import MultiProgress from '../multi-progress/multi-progress';
+import Button from '../button/button';
+import ErrorNotice from '$features/error-notice/error-notice';
+import ImageCdnRecommendation from '$features/image-size-analysis/image-cdn-recommendation/image-cdn-recommendation';
+import { recordBoostEvent, recordBoostEventAndRedirect } from '$lib/utils/analytics';
+import getIsaErrorSuggestion from '$lib/utils/get-isa-error-suggestion';
+import RefreshIcon from '$svg/refresh';
+import WarningIcon from '$svg/warning-outline';
+import {
+	type IsaCounts,
+	ISAStatus,
+	getReportProgress,
+	useIsaReport,
+	useImageAnalysisRequest,
+} from '$features/image-size-analysis';
+import { DataSyncProvider } from '@automattic/jetpack-react-data-sync-client';
+interface RecommendationsMetaProps {
+	isCdnActive: boolean;
+}
 
-export const RecommendationsMeta = () => (
-	<Notice level="info" title="Recommendations Meta" hideCloseButton={ true }>
-		<p>Implement RecommendationsMeta for ISA here.</p>
-	</Notice>
-);
+const _RecommendationsMeta: React.FC< RecommendationsMetaProps > = ( { isCdnActive } ) => {
+	const [ requestingReport, setRequestingReport ] = useState< boolean >( false );
+	const [ errorCode, setErrorCode ] = useState< number | undefined >( undefined );
+	const [ status, setStatus ] = useState< ISAStatus | undefined >( undefined );
+	const [ groups, setGroups ] = useState< Record< string, IsaCounts > >( {} );
+	const [ scannedPages, setScannedPages ] = useState< number >( 0 );
+	const [ totalIssues, setTotalIssues ] = useState< number >( 0 );
+	const [ errorMessage, setErrorMessage ] = useState< string | undefined >( undefined );
+	const [ errorSuggestion, setErrorSuggestion ] = useState< string | undefined >( undefined );
+	const [ waitNotice, setWaitNotice ] = useState< string | undefined >( undefined );
+	const [ showCDNRecommendation, setShowCDNRecommendation ] = useState< boolean >( false );
+	const [ { data: isaReport } ] = useIsaReport();
+	const requestNewReport = useImageAnalysisRequest();
+
+	useEffect( () => {
+		setStatus( isaReport?.status );
+		setGroups( isaReport?.groups || {} );
+		setScannedPages( scannedPagesCount( isaReport?.groups || {} ) );
```

Comment:
It seems like these states are used like variables. Can we skip the states and use variables?

```suggestion
	const [ totalIssues, setTotalIssues ] = useState< number >( 0 );
	const [ errorMessage, setErrorMessage ] = useState< string | undefined >( undefined );
	const [ errorSuggestion, setErrorSuggestion ] = useState< string | undefined >( undefined );
	const [ waitNotice, setWaitNotice ] = useState< string | undefined >( undefined );
	const [ showCDNRecommendation, setShowCDNRecommendation ] = useState< boolean >( false );
	const [ { data: isaReport } ] = useIsaReport();
	const requestNewReport = useImageAnalysisRequest();

	const status = isaReport?.status;
	const groups = isaReport?.groups || {};
	const scannedPages = scannedPagesCount( isaReport?.groups || {} )

	useEffect( () => {
```

Reply:
Possibly, will circle back on this. 👍